File: ReverseRowMatrix/Reverse.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 7/08/2024
    Program : Write a program which accept matrix and reverse the contents of each row. 
*/

//Required Libraries
package main;
import "fmt"

//Function to Reverse each row of the matrix 
func ReverseRow(arr[][] int,irow int,icol int) {
	var i int = 0;
	var j int = 0;

	//Logic to reverse each row of the matrix 
	fmt.Println("Updated matrix is as follows : ");
	for i = 0;i < irow;i++ {
		for j = icol - 1;j > -1 ;j-- {
			fmt.Printf("%d\t",arr[i][j]);
		}
		fmt.Printf("\n");
	}
}

//Main Function
func main() {
	var arr[][] int = nil;
	var rows int = 0;
	var cols int = 0;
	var i int = 0;
	var j int = 0;

	fmt.Println("Jay Ganesh....");

	//Taking input from user
	fmt.Print("Enter Number of Rows : ");
	fmt.Scan(&rows);

	fmt.Print("Enter Number of Columns : ");
	fmt.Scan(&cols);

	//Allocating memory for matrix
	arr = make([][]int,rows);

	for i = 0;i < rows;i++ {
		arr[i] = make([]int,cols);
	} 
	
	//Inserting elements in matrix 
	fmt.Println("Enter elements of matrix : ");
	for i = 0;i < rows;i++ {
		fmt.Printf("---Enter elements of %d row : ---\n",(i + 1));
		for j = 0;j < cols;j++ {
			fmt.Printf("Enter element no %d : \n",(j + 1));
			fmt.Scan(&arr[i][j]);
		}
	}

	//Printing elements of matrix
	fmt.Println("Elements of matrix are : ");
	for i = 0;i < rows;i++ {
		for j = 0;j < cols;j++ {
			fmt.Printf("%d\t",arr[i][j]);
		}
		fmt.Printf("\n");
	}

	//Call to ReverseRow function
	ReverseRow(arr,rows,cols);
}

File: ReverseRowMatrix/Reverse.js
/*
    Author : Pakshal Shashikant Jain
    Date : 7/08/2024
    Program : Write a program which accept matrix and reverse the contents of each row.
*/
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Function to Reverse each row of the matrix 
function ReverseRow(arr, rows, cols) {
    var i = 0;
    var j = 0;
    //Logic to reverse each row of the matrix
    console.log("Updated matrix is as follows : ");
    for (i = 0; i < rows; i++) {
        for (j = cols - 1; j > -1; j--) {
            process.stdout.write(arr[i][j] + "\t");
        }
        console.log();
    }
}
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//Main function
function main() {
    var arr;
    var row = 4;
    var col = 4;
    var i = 0;
    var j = 0;
    console.log("Jay Ganesh....");
    //Allocating memory for matrix 
    arr = new Array(row);
    for (i = 0; i < row; i++) {
        arr[i] = new Array(col);
    }
    //Inserting elements in matrix
    for (i = 0; i < row; i++) {
        for (j = 0; j < col; j++) {
            arr[i][j] = (i * col) + j + 10;
        }
    }
    //Printing Elements of matrix
    console.log("Elements of Matrix are : ");
    for (i = 0; i < row; i++) {
        for (j = 0; j < col; j++) {
            process.stdout.write(arr[i][j] + "\t");
        }
        console.log();
    }
    //Call to Reverse Row function
    ReverseRow(arr, row, col);
}
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//call to main function
main();
